drivers/pinctrl/Kconfig.xec

```diff
@@ -5,7 +5,7 @@ DT_COMPAT_ST_PINCTRL_XEC := microchip,xec-pinctrl
 
 config PINCTRL_MCHP_XEC
 	bool "Pin controller driver for MCHP XEC MCUs"
-	depends on SOC_SERIES_MEC172X
+	depends on SOC_FAMILY_MEC
 	default $(dt_compat_enabled,$(DT_COMPAT_ST_PINCTRL_XEC))
```

Comment:
```suggestion
	default y
	depends on DT_HAS_MICROCHIP_XEC_PINCTRL_ENABLED
```

(and delete L4)

Reply:
I think a rebase is needed since upstream already does this.

Reply:
after the rebase, I am getting compilation errors, not sure why, working on it.
The MEC15xx build is compiling xec_gpio_v2 driver meant for mec17xx

Reply:
the gpio_v2 driver was getting pulled due to the dts; fixed it in `mec1501hsz.dtsi`